### ISSUE TITLE
Xcode 9, Provisioning Profiles in simplest possible case doesn't work despite following docs

### ISSUE THREAD

**Issue by Rob117** [fastlane/fastlane#12549]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

bundle exec fast lane beta fails (message below). I have read every open issue on the problem. It seems to be an XCode 9 thing? Match appears to be working just fine, and my fastfile is very simple. I get the feeling I'm not writing something where I should be. I specified the provisioning profiles for release and debug in Xcode 9 in accordance with the docs.

**Fastfile**:
```
default_platform(:ios)

platform :ios do
  desc "Push a new beta build to TestFlight"
  lane :beta do
    ensure_git_status_clean
    increment_build_number(xcodeproj: "FastDeployPush.xcodeproj")
    commit_version_bump(xcodeproj: "FastDeployPush.xcodeproj")
    add_git_tag
    push_to_git_remote
    match(
      type: "appstore",
      git_url: "git@github.com:Rob117/m3prot_cert_test.git"
    )
    build_app(
      scheme: "FastDeployPush",
    )
    upload_to_testflight
  end
end
```
##### Complete output when running fastlane, including the stack trace and command used
installed via ruby gem
`bundle exec fast lane beta`

-- note -> The --capture_output command returns a rocket and a checkmark. :(

<details>
  <pre>[18:24:25]: ▸ Running script 'Bundle React Native code and images'
[18:24:28]: ▸ the transform cache was reset.
[18:24:45]: ▸ Touching FastDeployPush.app
[18:24:45]: ▸ Signing /Users/r-sherling/Library/Developer/Xcode/DerivedData/FastDeployPush-cjefjkvkhdneficbpwhhfjrtkysp/Build/Intermediates.noindex/ArchiveIntermediates/FastDeployPush/InstallationBuildProductsLocation/Applications/FastDeployPush.app
[18:24:54]: ▸ Touching FastDeployPush.app.dSYM
[18:24:55]: ▸ Archive Succeeded
[18:24:55]: Generated plist file with the following values:
[18:24:55]: ▸ -----------------------------------------
[18:24:55]: ▸ {
[18:24:55]: ▸   "provisioningProfiles": {
[18:24:55]: ▸     "com.m3career.FastDeployPush": "match AppStore com.m3career.FastDeployPush"
[18:24:55]: ▸   },
[18:24:55]: ▸   "method": "app-store",
[18:24:55]: ▸   "signingStyle": "manual"
[18:24:55]: ▸ }
[18:24:55]: ▸ -----------------------------------------
[18:24:55]: $ /usr/bin/xcrun /Users/r-sherling/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/fastlane-2.95.0/gym/lib/assets/wrap_xcodebuild/xcbuild-safe.sh -exportArchive -exportOptionsPlist '/var/folders/n6/p2zt93fd67l_p0cyzvkllqnh0000gp/T/gym_config20180518-47071-hlg5k8.plist' -archivePath /Users/r-sherling/Library/Developer/Xcode/Archives/2018-05-18/FastDeployPush\ 2018-05-18\ 18.22.40.xcarchive -exportPath '/var/folders/n6/p2zt93fd67l_p0cyzvkllqnh0000gp/T/gym_output20180518-47071-rrqxfj' 
rbenv detected, removing env variables
rbenv: no such command `shell'
+ xcodebuild -exportArchive -exportOptionsPlist /var/folders/n6/p2zt93fd67l_p0cyzvkllqnh0000gp/T/gym_config20180518-47071-hlg5k8.plist -archivePath '/Users/r-sherling/Library/Developer/Xcode/Archives/2018-05-18/FastDeployPush 2018-05-18 18.22.40.xcarchive' -exportPath /var/folders/n6/p2zt93fd67l_p0cyzvkllqnh0000gp/T/gym_output20180518-47071-rrqxfj
2018-05-18 18:24:55.939 xcodebuild[50011:2709699] [MT] IDEDistribution: -[IDEDistributionLogging _createLoggingBundleAtPath:]: Created bundle at path '/var/folders/n6/p2zt93fd67l_p0cyzvkllqnh0000gp/T/FastDeployPush_2018-05-18_18-24-55.938.xcdistributionlogs'.
2018-05-18 18:24:56.155 xcodebuild[50011:2709699] [MT] IDEDistribution: Step failed: <IDEDistributionSigningAssetsStep: 0x7fd91d7b3280>: Error Domain=IDEDistributionSigningAssetStepErrorDomain Code=0 "Locating signing assets failed." UserInfo={NSLocalizedDescription=Locating signing assets failed., IDEDistributionSigningAssetStepUnderlyingErrors=(
"Error Domain=IDEProvisioningErrorDomain Code=9 \"\"FastDeployPush.app\" requires a provisioning profile.\" UserInfo={NSLocalizedDescription=\"FastDeployPush.app\" requires a provisioning profile., NSLocalizedRecoverySuggestion=Add a profile to the \"provisioningProfiles\" dictionary in your Export Options property list.}"
)}
error: exportArchive: "FastDeployPush.app" requires a provisioning profile.

Error Domain=IDEProvisioningErrorDomain Code=9 ""FastDeployPush.app" requires a provisioning profile." UserInfo={NSLocalizedDescription="FastDeployPush.app" requires a provisioning profile., NSLocalizedRecoverySuggestion=Add a profile to the "provisioningProfiles" dictionary in your Export Options property list.}

** EXPORT FAILED **
[18:24:56]: Exit status: 70
[18:24:56]: No provisioning profile provided
[18:24:56]: Make sure to pass a valid provisioning for each required target
[18:24:56]: Check out the docs on how to fix this: https://docs.fastlane.tools/actions/gym/#export-options

+---------------+-------------------------+
|            Build environment            |
+---------------+-------------------------+
| xcode_path    | /Applications/Xcode.app |
| gym_version   | 2.95.0                  |
| export_method | app-store               |
| sdk           | iPhoneOS11.2.sdk        |
+---------------+-------------------------+

[18:24:56]: ▸     cd /Users/r-sherling/test_projects/FastDeployPush/ios
[18:24:56]: ▸     export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/Users/r-sherling/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/bin:/Users/r-sherling/.rbenv/versions/2.2.4/bin:/Users/r-sherling/homebrew/Cellar/rbenv/1.1.1/libexec:/Users/r-sherling/.rbenv/shims:/Users/r-sherling/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/munki:/Users/r-sherling/Library/Android/sdk/tools:/Users/r-sherling/Library/Android/sdk/platform-tools"
[18:24:56]: ▸     /usr/bin/touch -c /Users/r-sherling/Library/Developer/Xcode/DerivedData/FastDeployPush-cjefjkvkhdneficbpwhhfjrtkysp/Build/Intermediates.noindex/ArchiveIntermediates/FastDeployPush/BuildProductsPath/Release-iphoneos/FastDeployPush.app.dSYM
[18:24:56]: ▸ ** ARCHIVE SUCCEEDED **
[18:24:56]: 
[18:24:56]: ⬆️  Check out the few lines of raw `xcodebuild` output above for potential hints on how to solve this error
[18:24:56]: 📋  For the complete and more detailed error log, check the full log at:
[18:24:56]: 📋  /Users/r-sherling/Library/Logs/gym/FastDeployPush-FastDeployPush.log
[18:24:56]: 
[18:24:56]: Looks like fastlane ran into a build/archive error with your project
[18:24:56]: It's hard to tell what's causing the error, so we wrote some guides on how
[18:24:56]: to troubleshoot build and signing issues: https://docs.fastlane.tools/codesigning/getting-started/
[18:24:56]: Before submitting an issue on GitHub, please follow the guide above and make
[18:24:56]: sure your project is set up correctly.
[18:24:56]: fastlane uses `xcodebuild` commands to generate your binary, you can see the
[18:24:56]: the full commands printed out in yellow in the above log.
[18:24:56]: Make sure to inspect the output above, as usually you'll find more error information there
[18:24:56]: 
+------------------------------------+--------------------------------------------------+
|                                     Lane Context                                      |
+------------------------------------+--------------------------------------------------+
| DEFAULT_PLATFORM                   | ios                                              |
| PLATFORM_NAME                      | ios                                              |
| LANE_NAME                          | ios beta                                         |
| GIT_REPO_WAS_CLEAN_ON_START        | true                                             |
| BUILD_NUMBER                       | 10                                               |
| SIGH_PROFILE_TYPE                  | app-store                                        |
| MATCH_PROVISIONING_PROFILE_MAPPING | {"com.m3career.FastDeployPush"=>"match AppStore  |
|                                    | com.m3career.FastDeployPush"}                    |
+------------------------------------+--------------------------------------------------+
[18:24:56]: Error packaging up the application

+------+-------------------------+-------------+
|               fastlane summary               |
+------+-------------------------+-------------+
| Step | Action                  | Time (in s) |
+------+-------------------------+-------------+
| 1    | default_platform        | 0           |
| 2    | ensure_git_status_clean | 0           |
| 3    | increment_build_number  | 0           |
| 4    | commit_version_bump     | 0           |
| 5    | add_git_tag             | 0           |
| 6    | push_to_git_remote      | 8           |
| 7    | match                   | 22          |
| 💥   | build_app               | 143         |
+------+-------------------------+-------------+

[18:24:56]: fastlane finished with errors

[!] Error packaging up the application
</pre>
</details>

### Environment
<details>
  <pre><details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.6                                     |
| Ruby                        | 2.2.4                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.14.3 (Apple Git-98)           |
| Installation Source         | ~/.rbenv/versions/2.2.4/bin/fastlane        |
| Host                        | Mac OS X 10.12.6 (16G1212)                  |
| Ruby Lib Dir                | ~/.rbenv/versions/2.2.4/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2n  7 Dec 2017                  |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.2                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
default_platform(:ios)

platform :ios do
  desc "Push a new beta build to TestFlight"
  lane :beta do
    ensure_git_status_clean
    increment_build_number(xcodeproj: "FastDeployPush.xcodeproj")
    commit_version_bump(xcodeproj: "FastDeployPush.xcodeproj")
    add_git_tag
    push_to_git_remote
    match(
      type: "appstore",
      git_url: "git@github.com:Rob117/STUFF"
    )
    build_app(
      scheme: "FastDeployPush",
    )
    upload_to_testflight
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier("EMAIL") # The bundle identifier of your app
apple_id("EMAIL") # Your Apple email address

itc_team_id("STUFF") # iTunes Connect Team ID
team_id("STUFF") # Developer Portal Team ID

# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.95.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.2        |
| CFPropertyList            | 3.0.0        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.5        |
| xcodeproj                 | 1.5.9        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.3.2        |
| terminal-table            | 1.8.0        |
| plist                     | 3.4.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.6.4        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.8.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.62.0       |
| faraday                   | 0.15.1       |
| unf_ext                   | 0.0.7.5      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20180417 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.3        |
| gh_inspector              | 1.1.3        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.13.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.4.0        |
| bundler                   | 1.16.1       |
| faraday_middleware        | 0.12.2       |
| naturally                 | 2.1.0        |
| simctl                    | 1.6.4        |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.6       |
| emoji_regex               | 0.1.1        |
| json                      | 2.1.0        |
</details>

*generated on:* **2018-05-18**
</details></pre>
</details>

**Comment by Rob117** [comment]
@fastlane-bot yes, yes I have.

**Comment by Rob117** [comment]
Solved with #12309
